### PR TITLE
docs: Add guide on resuming training from weight-only checkpoints

### PR DESCRIPTION
**Pull request by Sikandar1310291**
This PR addresses a common point of confusion for users resuming training from checkpoints saved with `save_weights_only=True` (or `model.save_weights()`).

Many users assume that loading weights into a fresh model will not restore the optimizer state (like `iterations` for learning rate schedules). However, in Keras 3, the `.weights.h5` format *does* preserve this state, provided the model is compiled *before* loading weights.

I added a new subsection "Resuming training from weight-only checkpoints" to [guides/training_with_built_in_methods.py](cci:7://file:///c:/Users/ma516/OneDrive/Desktop/keras/guides/training_with_built_in_methods.py:0:0-0:0) that:
1. Clarifies that optimizer state is saved in weights files.
2. Explains the requirement to compile the model first.
3. Provides a concise code snippet demonstrating the correct resumption pattern.

### Related Issues
Relates to user confusion seen in discussions/issues about resuming learning rate schedules after interruption.

### CODE REVIEW

File: guides/training_with_built_in_methods.py [hunk 1]
```diff
@@ -1155,6 +1155,37 @@ def make_or_restore_model():
...
+    loss="sparse_categorical_crossentropy",
+    metrics=["sparse_categorical_accuracy"],
+)
+
+# 3. Load the weights
+# The optimizer state (iterations, etc.) is restored here automatically
+model.load_weights("checkpoint.weights.h5")
+
+# 4. Continue training
+model.fit(x_train, y_train, epochs=10)
```
[the remark this answers is not in the document]
Comment (Sikandar1310291) [reply guides/training_with_built_in_methods.py]: Thank you for catching this! You're absolutely right — the lr_schedule was undefined which would cause a NameError for users. I've pushed a fix that adds the schedule definition to make the example self-contained and runnable.

File: guides/training_with_built_in_methods.py [hunk 2]
```diff
@@ -1155,6 +1155,43 @@ def make_or_restore_model():
 ]
 model.fit(x_train, y_train, epochs=1, callbacks=callbacks)
 
+"""
+### Resuming training from weight-only checkpoints
+
+When using `ModelCheckpoint` with `save_weights_only=True` (or calling `model.save_weights()`),
```
Comment (hertschuh) [inline guides/training_with_built_in_methods.py]: In this location, nobody will find this guide.

Can you move it here: https://github.com/keras-team/keras/blob/master/keras/src/callbacks/model_checkpoint.py#L68

This way it will appear here: https://keras.io/api/callbacks/model_checkpoint/

**Comment by Sikandar1310291** [comment]
@Sikandar1310291 Great news! All test runs have passed successfully! 🎉

All 8 checks have been completed:
✅ CLA/Google
✅ Check the code format
✅ Run tests (JAX, NNX enabled = false)
✅ Run tests (JAX, NNX enabled = true)
✅ Run tests (NumPy)
✅ Run tests (OpenVINO)
✅ Run tests (TensorFlow)
✅ Run tests (PyTorch)

The PR is ready for review and merging. Thank you for your guidance and support throughout this contribution!

File: guides/training_with_built_in_methods.py (hunk 2 above)
Comment (Sikandar1310291) [reply guides/training_with_built_in_methods.py]: @hertschuh Thank you for the feedback! I've already moved the guide to the ModelCheckpoint docstring in commits 08ff714 and e4b6412. The guide has been removed from guides/training_with_built_in_methods.py and is now properly placed in the ModelCheckpoint docstring so it will be discoverable at https://keras.io/api/callbacks/model_checkpoint/ as you suggested. All tests have passed successfully! 🎉

File: keras/src/callbacks/model_checkpoint.py [hunk 3]
```diff
@@ -68,6 +68,43 @@ class ModelCheckpoint(MonitorCallback):
...
+
+    # 1. Create a fresh model instance
+    model = get_model()
+
+    # 2. Compile the model *before* loading weights
+    model.compile(
+        optimizer=keras.optimizers.RMSprop(learning_rate=lr_schedule),
+        loss="sparse_categorical_crossentropy",
+        metrics=["accuracy"],
+    )
```
Comment (hertschuh) [inline keras/src/callbacks/model_checkpoint.py:114]: Just for consistency and the avoid any confusion, can you copy these lines to lines 37-38?

File: keras/src/callbacks/model_checkpoint.py [hunk 4]
```diff
@@ -68,6 +68,43 @@ class ModelCheckpoint(MonitorCallback):
...
+
+    # 3. Load weights (optimizer state is restored automatically)
+    model.load_weights(checkpoint_filepath)
+
+    # 4. Continue training
+    model.fit(x_train, y_train, epochs=10)
```
Comment (hertschuh) [inline keras/src/callbacks/model_checkpoint.py:120]: I just noticed that lines 49 and 65 are incorrect. Can you change them from
```python
model.fit(epochs=EPOCHS, callbacks=[model_checkpoint_callback])
```
to
```python
model.fit(x_train, y_train, epochs=10, callbacks=[model_checkpoint_callback])
```
for correctness and consistency.

**Review by hertschuh — COMMENTED** [review]
Thanks for adding this!

Can I ask you to fix some pre-existing issues in that docstring?

File: keras/src/backend/__init__.py [hunk 5]
```diff
@@ -74,4 +74,69 @@ class name_scope(backend_name_scope):
 
 @keras_export("keras.device")
 def device(device_name):
+    """Context manager for backend-agnostic device placement.
```
Comment (hertschuh) [inline keras/src/backend/__init__.py]: Oh, this file should be in the other PR, can you revert here?
